packages/opds-web-client/src/components/Root.tsx

```diff
@@ -81,6 +81,8 @@ export interface RootProps extends StateProps {
   closeErrorAndHideAuthForm?: () => void;
   setPreference?: (key: string, value: string) => void;
   allLanguageSearch?: boolean;
+  router?: {};
+  pathFor?: () => {};
```

Comment:
how did these end up in the props?

Reply:
Tslint complained. In some cases I wanted to just fix the issues and make sure everything was working before going back and fixing it.

Reply:
I'd still like to investigate this.

Reply:
Okay, I see what was wrong. The `contextTypes` were being passed the `RootProps` - `React.ValidationMap<RootProps>`. It should be the `NavigateContext` - `React.ValidationMap<NavigateContext>`.